guides/warning-for-head-and-tail.md

```diff
@@ -13,10 +13,10 @@ This is not a breaking change, unless a package forces `-Werror`.
 
 If you don't want to change any Haskell code and want to carry on with using `Data.List.{head,tail}`:
 
-1. If you use `-Werror`, set `-Wwarn=x-partial` to downgrade this area of compiler messages from errors to warnings, so that your build can succeed.
+1. If you use `-Werror`, set `-Wwarn=x-partial` to downgrade this area of compiler messages from errors to warnings, so that your build can succeed. Note that `x-partial` is a new group of warnings, introduced in GHC 9.8, so you might need additionally set `-Wno-unrecognised-warning-flags` or `-Wwarn=unrecognised-warning-flags` to maintain backward compatibility with older GHC releases.
```

Comment:
grammar nit: 

```suggestion
1. If you use `-Werror`, set `-Wwarn=x-partial` to downgrade this area of compiler messages from errors to warnings, so that your build can succeed. Note that `x-partial` is a new group of warnings, introduced in GHC 9.8, so you might need to additionally set `-Wno-unrecognised-warning-flags` or `-Wwarn=unrecognised-warning-flags` to maintain backward compatibility with older GHC releases.
```